cmd/asset-syncer/utils_test.go

```diff
@@ -1235,3 +1301,45 @@ func Test_filterCharts(t *testing.T) {
 		})
 	}
 }
+
+func Test_compareURLDomains(t *testing.T) {
+	tests := []struct {
+		name     string
+		url1     string
+		url2     string
+		expected bool
+	}{
+		{
+			"it returns false if a url is malformed",
+			"abc",
+			"https://bitnami.com/assets/stacks/airflow/img/airflow-stack-220x234.png",
+			false,
+		},
+		{
+			"it returns false if they are different subdomains",
+			"https://charts.bitnami.com/bitnami/index.yaml",
+			"https://bitnami.com/assets/stacks/airflow/img/airflow-stack-220x234.png",
+			false,
+		},
+		{
+			"it returns false if they are under different subdomains",
+			"https://bitnami.com/bitnami/index.yaml",
+			"https://charts.bitnami.com/assets/stacks/airflow/img/airflow-stack-220x234.png",
+			false,
+		},
```

Comment:
These two tests appear to be testing the same thing? Also, this is interesting - the subdomain. Is this the same behaviour in the helm patch? Asking because it's much more likely that I'd have my assets on a subdomain of the repo... but fine either way.

Reply:
They are using the same approach (below), so I think we should stick close to it. If a user happens to really need to pass the credentials down to a different subdomain, we may consider adding a `--allowed-pass-credentials-subdomains` flag

```
(u1.Scheme == u2.Scheme && u1.Host == u2.Host)
```

Side note: we were using `.Hostname()`, which yields the `Host` value without ports. Having noticed they also compare the ports, I've changed the PR's code to match their appraoch.